src/MiniExcel/OpenXml/ExcelOpenXmlSheetWriter.cs

```diff
@@ -143,137 +295,7 @@ private void CreateSheetXml(object value, string sheetPath)
-                    writer.Write($@"<x:dimension ref=""{GetDimensionRef(maxRowIndex, maxColumnIndex)}""/>");
-
-                    //cols:width
-                    var ecwProp = props?.Where(x => x?.ExcelColumnWidth != null).ToList();
-                    if (ecwProp != null && ecwProp.Count > 0)
-                    {
-                        writer.Write($@"<x:cols>");
-                        foreach (var p in ecwProp)
-                        {
-                            writer.Write($@"<x:col min=""{p.ExcelColumnIndex + 1}"" max=""{p.ExcelColumnIndex + 1}"" width=""{p.ExcelColumnWidth}"" customWidth=""1"" />");
-                        }
-                        writer.Write($@"</x:cols>");
-                    }
-
-                    //header
-                    writer.Write($@"<x:sheetData>");
-                    var yIndex = 1;
-                    var xIndex = 1;
-                    if (_printHeader)
-                    {
-                        var cellIndex = xIndex;
-                        writer.Write($"<x:row r=\"{yIndex}\">");
-
-                        foreach (var p in props)
-                        {
-                            if (p == null)
-                            {
-                                cellIndex++; //reason : https://github.com/shps951023/MiniExcel/issues/142
-                                continue;
-                            }
-
-                            var r = ExcelOpenXmlUtils.ConvertXyToCell(cellIndex, yIndex);
-                            WriteC(writer, r, columnName: p.ExcelColumnName);
-                            cellIndex++;
-                        }
-
-                        writer.Write($"</x:row>");
-                        yIndex++;
-                    }
-
-                    // body
-                    if (mode == "IDictionary<string, object>") //Dapper Row
-                        GenerateSheetByColumnInfo<IDictionary<string, object>>(writer, value as IEnumerable, props, xIndex, yIndex);
-                    else if (mode == "IDictionary") //IDictionary
-                        GenerateSheetByColumnInfo<IDictionary>(writer, value as IEnumerable, props, xIndex, yIndex);
-                    else if (mode == "Properties")
-                        GenerateSheetByColumnInfo<object>(writer, value as IEnumerable, props, xIndex, yIndex);
-                    else
-                        throw new NotImplementedException($"Type {type.Name} & genericType {genericType.Name} not Implemented. please issue for me.");
-                    writer.Write("</x:sheetData>");
-                    if (_configuration.AutoFilter)
-                        writer.Write($"<x:autoFilter ref=\"A1:{ExcelOpenXmlUtils.ConvertXyToCell(maxColumnIndex, maxRowIndex == 0 ? 1 : maxRowIndex)}\" />");
-                    writer.Write("<x:drawing  r:id=\"drawing" + currentSheetIndex + "\" /></x:worksheet>");
+                    GenerateSheetByEnumerable(writer, value as IEnumerable, out genericType);

```

Comment:
`out genericType` seems unnecessary here.
Moreover, the `genericType` variable in line 306 is always null and will result in an NulllReferenceExceptoin being thrown - you can correct this by the way

Reply:
Yeah I had not thought of that. I removed it and used the full name of the input value instead